File: Server/src/main/java/com/webdev/jobify/model/Certificate.java
package com.webdev.jobify.model;


import com.fasterxml.jackson.annotation.JsonFormat;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;

@Entity
@Table(name = "certificates")
public class Certificate implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false)
    private String school;

    @Column(nullable = false)
    private String degree;

    private String field;

    @JsonFormat(pattern = "dd-MM-yyyy")
    private Date startDate;

    @JsonFormat(pattern = "dd-MM-yyyy")
    private Date endDate;

    private float grade;

    @ManyToOne
    Employee employee;

    @Column(nullable = false)
    private boolean hidden;

    public Certificate() {}


    public Certificate(Long id, String school, String degree, String field, Date startDate, Date endDate, float grade, Employee employee) {
        this.id = id;
        this.school = school;
        this.degree = degree;
        this.field = field;
        this.startDate = startDate;
        this.endDate = endDate;
        this.grade = grade;
        this.employee = employee;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getSchool() {
        return school;
    }

    public void setSchool(String school) {
        this.school = school;
    }

    public String getDegree() {
        return degree;
    }

    public void setDegree(String degree) {
        this.degree = degree;
    }

    public String getField() {
        return field;
    }

    public void setField(String field) {
        this.field = field;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public float getGrade() {
        return grade;
    }

    public void setGrade(float grade) {
        this.grade = grade;
    }

    public Employee getEmployee() {
        return employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }

    public boolean isHidden() {
        return hidden;
    }

    public void setHidden(boolean hidden) {
        this.hidden = hidden;
    }
}

File: Client/src/app/_models/certificate/certificate.ts
import * as globals from "../../globals"

export class Certificate {

    id: number;
    school: string;
    degree: string;
    field: string;
    startDate: string;
    endDate: string;
    grade: number;
    hidden: boolean;

    constructor(id: number, school: string, degree: string, field: string,
        startDate: string, endDate: string, grade: number, hidden: boolean) {
        this.id = id;
        this.school = school;
        this.degree = degree;
        this.field = field;
        this.startDate = startDate;
        this.endDate = endDate;
        this.grade = grade;
        this.hidden = hidden;
        
      }

      public beautify() {
        if(this.startDate != null) {
            let month = this.startDate.split('-')[1];
            let year = this.startDate.split('-')[0];
            this.startDate = ' ' + globals.findMonth(month) + ' ' + year;
        }
          
        if(this.endDate != null) {
          let month = this.endDate.split('-')[1];
          let year = this.endDate.split('-')[0];
          this.endDate = ' - ' + globals.findMonth(month) + ' ' + year;
        }

      }
      

}
